packages/vscode-extension/src/commands/CompareFile.ts

```diff
@@ -0,0 +1,97 @@
+
+	constructor() {
+		super(CompareFile.COMMAND_NAME);
+	}
+
+	protected validateBeforeExecute() {
+		const superValidation = super.validateBeforeExecute();
+		if (!superValidation.valid) {
+			return superValidation;
+		}
+
+		if (!this.activeFileIsUnderUnrestrictedFolder()) {
+			return this.unsuccessfulValidation(this.translationService.getMessage(IMPORT_FILES.ERROR.NOT_ALLOWED_FOLDER));
+		}
+
+		return this.successfulValidation();
+	}
+
+	protected async execute() {
+		const activeFilePath = this.activeFile!;
+		const tempFolderPath = fs.mkdtempSync(path.join(os.tmpdir(), CompareFile.TEMP_FOLDER_PREFIX));
+
+		this.copyManifestFileToTempFolder(tempFolderPath);
+		this.copyProjectJsonToTempFolder(tempFolderPath);
+		const activeFileRelativePath = activeFilePath.split(this.getFileCabinetFolderPath())[1]?.replace(/\\/g, '/');
+		const importFilePath = this.copyActiveFileToTempFolder(tempFolderPath, activeFilePath, activeFileRelativePath);
+
+		const selectedFilesPaths = [];
+		selectedFilesPaths.push(activeFileRelativePath);
+
+		const commandArgs = {
+			paths: selectedFilesPaths,
+			excludeproperties: 'true',
+		};
+
+		const commandActionPromise = this.runSuiteCloudCommand(commandArgs, tempFolderPath);
+		this.messageService.showStatusBarMessage(this.translationService.getMessage(COMPARE_FILE.COMPARING_FILE), true, commandActionPromise);
+		const actionResult = await commandActionPromise;
+		if (actionResult.status === actionResultStatus.SUCCESS && actionResult.data) {
+			const compareWindowTitle =
+				this.translationService.getMessage(COMPARE_FILE.COMPARE_FILE_WITH_ACCOUNT_FILE) + ' - ' + path.basename(activeFilePath);
+			vscode.commands.executeCommand('vscode.diff', vscode.Uri.file(activeFilePath), vscode.Uri.file(importFilePath), compareWindowTitle);
+		} else {
+			this.messageService.showCommandError();
+		}
+	}
+
+	protected async getSelectedFiles(): Promise<string[] | undefined> {
+		//Required but not used since execute is overwritten in this class.
+		return undefined;
+	}
+
+	private activeFileIsUnderUnrestrictedFolder(): boolean {
+		const activeFileRelativePath = this.activeFile?.replace(this.getFileCabinetFolderPath(), '').replace(/\\/g, '/');
+		return UNRESTRICTED_FOLDERS.some((unrestricedPath) => activeFileRelativePath?.startsWith(unrestricedPath));
+	}
+
+	private getFileCabinetFolderPath(): string {
+		return path.join(this.getProjectFolderPath(), ApplicationConstants.FOLDERS.FILE_CABINET);
+	}
+
+	private copyActiveFileToTempFolder(tempFolderPath: string, activeFilePath: string, activeFileRelativePath: string): string {
+		const importFileParentFolderPath = path.join(tempFolderPath, ApplicationConstants.FOLDERS.FILE_CABINET, path.dirname(activeFileRelativePath));
+		fs.mkdirSync(importFileParentFolderPath, { recursive: true });
+		const importFilePath = path.join(importFileParentFolderPath, path.basename(activeFilePath));
+		fs.copyFileSync(activeFilePath, importFilePath);
```

Comment:
Final question. Again, I may have not understood the full behavior.

Do we need to copy the original File to the temporary directory? I think it's no needed since the file:import command it's going to import it to the temp directory anyways.

Reply:
Good catch. No we don't. I pushed changes to remove copying of the original file.